content/docs/guide-development.md

```diff
@@ -75,6 +75,16 @@ It is very likely that your analysis workflow needs to access astronomical data
 Of course, relying on external data archives and data access services may pose a problem for reproducibility of results and potentially for the service operational stability. The external services may have downtime period, they may be upgraded and change their API etc. We leave it to the developer to make sure the requests to external services are operational and reproducible. We encourage the developers to supply tests that will be automatically executed from time to time during service operations and are supposed to always yield exactly the same results. If this is not so, this may signal a problem with non-availability of some external services, in this cases you, as the service developer, will be alerted and invited to investigate the origin of the problem.  It may also happen that the notebook would not produce the exactly
 the same result every time but still be reproducible (see motivation on [the difference between reproducibility and repeatability](https://github.com/volodymyrss/reproducibility-motivation/)). 
 
+### Handling exceptions
+
+It can happen that your analysis workflow is expected to produce no data products in some cases, for example, if there is no data for a specified source and time interval, if the parameters specified by the user have wrong format, or in other "exceptions"  In this case, it would be good to inform the user what happened. This can be done using the raise `RuntimeError()` method directly in the notebook, as shown below: 
+
+![image](tmp21.png)
+
+Don't worry if you do not succeed to foresee all possible exceptions at the initial development stage. If unforeseen exceptions would occur when the service is already deployed and availabel to users, each time an unforeseen exception occurs, you will be notified and invided to patch your notebook to handle this excpetion (perhaps arring a new `RuntimeError()` case in the appropriate cell).
```

Comment:
This is not implemented. It requires some changes to the plugin/dispatcher code. Some specific exception instead of RuntimeError needs to be introduced for "expected" cases in oda_api (there is an [issue](https://github.com/oda-hub/dispatcher-plugin-nb2workflow/issues/54) for this). Then, if the exception is not this special one, the email must be sent by dispatcher. Currently, dispatcher knows nothing about the authorship of the backend (btw, from bot's pov "author" is currently the one who made the last commit into master, should other commiters also be considered?).

Reply:
yes we should consider all commits and maybe also separately marked maintainer. https://github.com/oda-hub/dispatcher-plugin-nb2workflow/issues/70